sdk/search/azure-search/azure/search/_index/_queries.py

```diff
@@ -0,0 +1,98 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import TYPE_CHECKING
+
+from ._generated.models import AutocompleteRequest, SearchRequest, SuggestRequest
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Any, List, Type, Union
+
+
+class _QueryBase(object):
+
+    _request_type = (
+        None
+    )  # type: Union[Type[AutocompleteRequest], Type[SearchRequest], Type[SuggestRequest]]
+
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        self._request = self._request_type(**kwargs)  # pylint:disable=not-callable
+
+    def __repr__(self):
+        # type: () -> str
+        return "<{} [{}]>".format(self.__class__.__name__, self._request.search_text)[
+            :1024
+        ]
+
+    def filter(self, expression):
+        # type: (str) -> None
+        """Add a `filter` expression for the search results.
+
+        :param expression: An ODate expression of for the query filter.
+        :type expression: str
+        """
+        self._request.filter = expression
+
+    @property
+    def request(self):
+        """The service request for this operation.
+
+        """
+        return self._request
+
+
+class AutocompleteQuery(_QueryBase):
+
+    _request_type = AutocompleteRequest
+
+    __doc__ = AutocompleteRequest.__doc__
+
+
+class SearchQuery(_QueryBase):
```

Comment:
Does it make sense to have a constructor that takes optional `select`, `order_by` and a list of `filter` tuples? 

E.g. 
```
q  = SearchQuery(select=['a', 'b'], order_by=['b desc'])
```
?

Or are we just adding too many options on how to create this puppy...